test/hover.test.ts

```diff
@@ -40,7 +40,22 @@ describe('Hover Tests', () => {
     languageService.deleteSchema(SCHEMA_ID);
   });
 
-  function parseSetup(content: string, position): Promise<Hover> {
+  /**
+   * Generates hover information for the given document and caret (cursor) position.
+   * @param content The content of the document.
+   * @param position The position of the caret in the document.
+   * Alternatively, `position` can be omitted if the caret is located in the content using `|` bookends.
+   * For example, `content = 'ab|c|d'` places the caret over the `'c'`, at `position = 2`
+   * @returns An instance of `Hover`.
+   */
+  function parseSetup(content: string, position?: number): Promise<Hover> {
+    // console.log(`was: len: ${content.length}, content: "${content}", str: "${content.substring(position)}"`);
+    if (typeof position === 'undefined') {
+      position = content.search(/\|[^]\|/); // | -> any char including newline -> |
+      content = content.substring(0, position) + content.substring(position + 1, position + 2) + content.substring(position + 3);
```

Comment:
How about moving the above 2 lines to its own function in `test\utils\testHelper.ts`? Also as with others, what to do if caret is not found.

Reply:
Done